11_FFTBloom/app_resources/image_fft_first_axis.hlsl

```diff
@@ -72,21 +72,23 @@ struct PreloadedFirstAxisAccessor : PreloadedAccessorMirrorTradeBase
 		// Storing even elements of NFFT is storing the bitreversed lower half of DFT - see readme
 		for (uint32_t localElementIndex = 0; localElementIndex < ElementsPerInvocation; localElementIndex += 2)
 		{
+			complex_t<scalar_t> lo, hi;
 			// First element of 0th thread has a special packing rule
-			if (!workgroup::SubgroupContiguousIndex() && !localElementIndex)
+			if (!localElementIndex && !workgroup::SubgroupContiguousIndex())
 			{
-				complex_t<scalar_t> packedZeroNyquistLo = { preloaded[0].real(), preloaded[1].real() };
-				complex_t<scalar_t> packedZeroNyquistHi = { preloaded[0].imag(), preloaded[1].imag() };
-				storeColMajor(0, packedZeroNyquistLo, packedZeroNyquistHi);
+				lo.real(preloaded[0].real());
+				lo.imag(preloaded[1].real());
+				hi.real(preloaded[0].imag());
+				hi.imag(preloaded[1].imag());
 			}
 			else
 			{
-				complex_t<scalar_t> lo = preloaded[localElementIndex];
-				complex_t<scalar_t> hi = getDFTMirror<sharedmem_adaptor_t>(localElementIndex, adaptorForSharedMemory);
+				lo = preloaded[localElementIndex];
+				hi = getDFTMirror<sharedmem_adaptor_t>(localElementIndex, adaptorForSharedMemory);
```

Comment:
ekm, does `getDFTMirror` use a `workgroup::shuffle` or any `barrier()` call?

Reply:
It's a workgroup shuffle yeah. I'm rewriting it to do unpavking on load and it becomes a subgroup shuffle

Reply:
yes it does, I'm surprised you haven't deadlocked your GPU with this.

Reply:
What could deadlock here?

Reply:
> What could deadlock here?

depends on the particular GPUs implementation, but logically speaking a barrier could be a single special counter that counts how many invocations have arrived at it and prevents the arrived invocations from making progress until counter value equals workgroup size.

On actual implementations this could be done differently which might be why you didn't deadlock